### PR TITLE
Feature/home and login

### PR DESCRIPTION
**Pull request by Riky5**
As a user,
so that I can list a space,
I would like to login.

Created 
- homepage layout with login button
- login page

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -1,12 +1,41 @@
...
-    'Hello world!'
+    erb :index
+  end
+
+  get '/login' do
+    erb :login
+  end
+
+  post '/user_details' do
+    session[:username] = params[:username]
```
Comment (chris-clement) [inline app.rb:27]: are sessions preferable to instance variables? Would be interested to understand if there is a benefit of using sessions here

File: database_connection_setup.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+require './lib/database_connection'
```
Comment (chris-clement) [inline database_connection_setup.rb:3]: we set up something very similar. called setup_database.rb. We should ensure we only merge using one of these

File: lib/database_connection.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+require 'pg'
```
Comment (chris-clement) [inline lib/database_connection.rb:3]: we have this file too. Need to ensure we only include one

File: database_connection_setup.rb (hunk 2 above)
Comment (Riky5) [reply database_connection_setup.rb:3]: sure, we can discuss and decide which one to keep. Probably yours.

File: spec/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require 'user'
+
+describe User do
+  describe '.valid' do
+    it 'checks password and username is correct' do
```
Comment (chris-clement) [inline spec/user_spec.rb:7]: might be slightly neater to use context here. So context 'Password and username are correct' then under that it "returns true". context 'Password and username are inccorrect', under that it 'returns false'

**Review by chris-clement — COMMENTED** [review]
Looking good! A few comments from me

File: spec/user_spec.rb (hunk 4 above)
Comment (Riky5) [reply spec/user_spec.rb:7]: good advice, I need to get more used to using context 😅

File: public/index.css [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+#space-container {
+  max-width: 500px;
+  width: 100%;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  margin: auto;
+  border: 1px solid black;
+  padding: 1rem;
+}
```
Comment (Hereloss) [inline public/index.css:40]: Really liked all the styling for these pages - we'll definately have to add it to the other bits too I think!

**Review by Hereloss — COMMENTED** [review]
No real comments from me - I think we can get this merged in!

File: app.rb (hunk 1 above)
Comment (sabrinalord) [reply app.rb:27]: i read that session might slow things down if we had lots of users logged in as would take up memory, prob not an issue for our purposes but worth considering

**Pull request merged**